docs/kb/semgrep-ci/github-repository-rulesets-semgrep.md

```diff
@@ -0,0 +1,75 @@
+
+To make the secret available to all repositories that use this workflow, create the secret as an [organization secret](https://docs.github.com/en/enterprise-cloud@latest/actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-an-organization).
+
+If you prefer to not to add the secret as an organization secret, you should separately determine how to make the value available to the workflow when it runs on the target repositories.
+
+## Creating the organization ruleset
+
+To create the ruleset:
+
+1. Go to your GitHub organization page and click **Settings**.
+2. Under **Code, planning, and automation**, click **Repository** and then **Repository rulesets**.
+    ![Repository rulesets](/img/kb/semgrep-workflow-repository-rulesets.png).
+3. Click **New branch ruleset**.
+4. Configure the enforcement status, bypass list, target repositories, and target branches based on your organization policies.
+5. Under **Branch protections**, check the box **Require workflows to pass before merging**.
+6. Click **Add workflow**.
+7. In the **Add required workflow** modal, select the repository where you placed the Semgrep workflow.
+8. Then, select the branch, tag, or commit to use.
+9. In the **Pick a workflow file** field, click and select the Semgrep workflow you created in [Setting up the central Semgrep scan workflow](#setting-up-the-central-semgrep-scan-workflow).
+10. Click **Add workflow**.
+    ![Require workflows to pass before merging](/img/kb/semgrep-workflow-require-pass.png)
+11. Click **Create** to create the ruleset.
+
+You can refer to GitHub's [Creating rulesets for repositories in your organization](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-organization-settings/creating-rulesets-for-repositories-in-your-organization) for more general guidance on creating a ruleset for your organization.
+
+## Verify by creating a pull request
+
+After setup, hen you create a pull request in a repository where this ruleset applies, the Semgrep workflow runs for the PR even if there is no `.github/workflows/semgrep.yml` file in that repository.
+
+The required workflow allows merge if the scan is successful, or blocks the pull request if the scan has blocking findings.
+
+![Example PR with successful required workflow](/img/kb/semgrep-workflow-pr-example.png)
+
+## Limitations
+
+Workflows required by repository rulesets are only triggered by `pull_request` or `merge_group` events. When triggered for a pull request, Semgrep runs a [diff-aware scan] and only scans changed files. Therefore, to run a full scan for your organization's repositories (scan all files), you would need to choose a different approach, such as [reusable workflows](/docs/kb/semgrep-ci/github-reusable-workflows-semgrep).
```

Comment:
Are we, in effect, recommending that users use both reusable workflows AND repository rulesets to have full coverage (both full scans and diff scans)? Put another way, what is the ideal path for the user to take when implementing Semgrep scans for 100s or 1000s of repositories?

Reply:
(Also, I think the [diff-aware] scan is meant to be a link?

Reply:
yes totally was, ty for the reminder